File: scripts/migrate-v1.ts
import fs from "fs";

const oldData = JSON.parse(fs.readFileSync(process.argv[2], "utf-8"));

const newData = {
  version: "2.0",
  screens: [
    {
      id: "default",
      name: "Screen 1",
      components: oldData.components,
    },
  ],
  currentScreenId: "default",
  entities: oldData.entities.map(
    (entity: { name: string; properties: string[] }) => ({
      name: entity.name,
      properties: entity.properties.map((prop: string) => ({
        name: prop,
        type: "string" as const,
      })),
    })
  ),
};

console.log(JSON.stringify(newData, null, 2));

File: scripts/migrate_v1.py
#!/usr/bin/env python3
import json
import sys

with open(sys.argv[1], "r") as f:
    old_data = json.load(f)

new_data = {
    "version": "2.0",
    "screens": [
        {
            "id": "default",
            "name": "Screen 1",
            "components": old_data["components"],
        }
    ],
    "currentScreenId": "default",
    "entities": [
        {
            "name": entity["name"],
            "properties": [
                {"name": prop, "type": "string"} for prop in entity["properties"]
            ],
        }
        for entity in old_data["entities"]
    ],
}

print(json.dumps(new_data, ensure_ascii=False, indent=2))
